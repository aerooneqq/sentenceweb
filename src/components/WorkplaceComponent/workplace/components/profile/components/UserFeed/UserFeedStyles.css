@media screen and (min-width: 400px){ 
    #userFeedContainer{ 
        height: 450px;
        width: 400px;
        background: transparent;
        margin-bottom: 30px;
    }
    
    #userFeedInnerCont{ 
        height: 410px;
        width: 100%;
    
        background:#f9f9f9;
    
        border-style: none; 
        border-radius: 10px;
    }
    
    #userFeedInnerScroll{ 
        width: 100%;
        height: 360px;
    
        overflow-y: auto;
    
        background:#f9f9f9;
        border-style: none; 
        border-top-left-radius: 10px; 
        border-top-right-radius: 10px;
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        width: 3px;
    }
    
    #userFeedInnerScroll::-webkit-scrollbar-thumb{
        background:rgba(255,110,74, 0.5) !important; 
        border-radius: 2px; 
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        height: 0px;
    }   
    
    #userFeedInputCont{ 
        width: 100%;
        height: 53px;
    
        background:white;
    
        border-style: none ; 
        border-width: 1px;
        border-color: #F97646;
        border-bottom-left-radius: 10px;
        border-bottom-right-radius: 10px;
    
        display: flex; 
        flex-direction: row; 
        justify-content: left;
        align-items: center;
    }
    
    #userFeedTextArea{ 
        width:300px;
        height: 53px;
    
        border-style: solid; 
        border-bottom-left-radius: 10px;
        border-width: 1px; 
        border-color: #F97646;
    
        font-family: Arial; 
        font-size: 15px;
    
        padding: 5px;
        margin-top: 14px;
    
        resize: none;
    }
    
    #userFeedTextArea:focus{ 
        outline: none;
    }
    
    #sendFeedBtn{ 
        margin-top: 14px;
    
        width: 100px;
        height: 65px;
    
        background: white; 
        
        font-family: Arial; 
        font-size: 18px;
        color: #F97646;
    
        border-style: solid; 
        border-width: 1px;
        border-left-width: 0px; 
        border-color: #F97646;
        border-bottom-right-radius: 10px;
    }
    
    #sendFeedBtn:focus{ 
        outline: none;
    }
    
    #sendFeedBtn:hover{ 
        background: #F97646;
        color: white;
    }
}

@media screen and (max-width: 400px){ 
    #userFeedContainer{ 
        height: 450px;
        width: 300px;
    
        margin-top: 10px;
        margin-left: -15px;

        background: transparent;
    }
    
    #userFeedInnerCont{ 
        height: 410px;
        width: 100%;
    
        background:#f9f9f9;
    
        border-style: none; 
        border-radius: 10px;
    }
    
    #userFeedInnerScroll{ 
        width: 100%;
        height: 360px;
    
        overflow-y: auto;
    
        background:#f9f9f9;
        border-style: none; 
        border-top-left-radius: 10px; 
        border-top-right-radius: 10px;
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        width: 3px;
    }
    #userFeedInnerScroll::-webkit-scrollbar-thumb{
        background: #F97646;
        border-radius: 2px; 
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        height: 0px;
    }   
    
    #userFeedInputCont{ 
        width: 100%;
        height: 53px;
    
        background:white;
    
        border-style: none ; 
        border-width: 1px;
        border-color: #F97646;
        border-bottom-left-radius: 10px;
        border-bottom-right-radius: 10px;
    
        display: flex; 
        flex-direction: row; 
        justify-content: left;
        align-items: center;
    }
    
    #userFeedTextArea{ 
        width:300px;
        height: 53px;
    
        border-style: solid; 
        border-bottom-left-radius: 10px;
        border-width: 1px; 
        border-color: #F97646;
    
        font-family: Arial; 
        font-size: 15px;
    
        padding: 5px;
        margin-top: 14px;
    
        resize: none;
    }
    
    #userFeedTextArea:focus{ 
        outline: none;
    }
    
    #sendFeedBtn{ 
        margin-top: 14px;
    
        width: 100px;
        height: 65px;
    
        background: white; 
        
        font-family: Arial; 
        font-size: 18px;
        color: #F97646;
    
        border-style: solid; 
        border-width: 1px;
        border-left-width: 0px; 
        border-color: #F97646;
        border-bottom-right-radius: 10px;
    }
    
    #sendFeedBtn:focus{ 
        outline: none;
    }
    
    #sendFeedBtn:hover{ 
        background: #F97646;
        color: white;
    }
}
