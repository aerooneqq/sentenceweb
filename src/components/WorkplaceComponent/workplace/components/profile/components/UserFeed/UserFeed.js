import React, {Component, lazy, Suspense} from "react"

//Styles
import "./UserFeedStyles.css"

//Services
import UserFeedService from "../../../../../../../services/UserServices/UserFeedService";
import ProfileHeader from "../ProfileHeader/ProfileHeader";

//App messages
import {alertAppMessage} from "../../../../../../ApplicationMessage/ApplicationMessageManager";

//Components
const UserAtomFeed = lazy(()=>import("./UserAtomFeed/UserAtomFeed"));
const Loader = lazy(() => import("../../../../../../loader/Loader"));

export default class UserFeed extends Component { 

    constructor(props) { 
        super(props)

        this.state = { 
            component: <Loader message = "Loading feed..." />,
            textAreaValue: ""
        };

        this.userFeedService = new UserFeedService(localStorage.getItem("token"));

        this.insertNewPost = this.insertNewPost.bind(this);
        this.onTextAreaValueChange = this.onTextAreaValueChange.bind(this);
        this.uploadUserFeed = this.uploadUserFeed.bind(this);
        this.onTextAreaKeyDown = this.onTextAreaKeyDown.bind(this);

        this.uploadUserFeed();
    }

    /**
     * Deletes all repeated entries of elements
     * @param {Array of numbers} array 
     */
    _clearArray(array) { 
        var seen = {};
        var out = [];
        var len = array.length;
        var j = 0;

        for(var i = 0; i < len; i++) {
             var item = array[i];
             if(seen[item] !== 1) {
                   seen[item] = 1;
                   out[j++] = item;
             }
        }

        return out
    }

    uploadUserFeed() { 
        this.userFeedService.getUserFeed(localStorage.getItem("token"))
            .then(res => { 
                let atomFeeds = res.data.map(f => {
                    return  (
                        <UserAtomFeed date = {f.publicationDate}
                                      message = {f.message}
                                      photo = {f.photo}
                                      authorName = {f.name + " " + f.surname}/>
                    );
                });

                atomFeeds.reverse();

                this.setState({
                    component: atomFeeds,
                    textAreaValue: ""
                }); 
            }).catch(er => {
                if (er.response) { 
                    alertAppMessage(er.reponse.data, "error");
                }
                else { 
                    alertAppMessage("Error occured while getting your feed", "error");
                }
            });
    }

    insertNewPost() { 
        this.setState({ 
            component: <Loader message = "Inserting your message" />
        });

        let message = this.state.textAreaValue;

        this.userFeedService.insertUserPost(message).then(() => { 
                alertAppMessage("The message was posted!", "success");
                this.uploadUserFeed();
            }).catch(er => { 
                if (er.response) { 
                    alertAppMessage(er.reponse.data, "error");
                }
                else { 
                    alertAppMessage("The error occured while inserting your message", "error");
                }
            });

    }

    onTextAreaValueChange(event) { 
        this.setState({ 
            textAreaValue: event.target.value
        });
    }

    onTextAreaKeyDown(event) { 
        if (event.keyCode === 13) { 
            this.insertNewPost();
        }
    }

    render() { 
        return(
            <div id = "userFeedContainer">
                <ProfileHeader header = "Feed" /> 
                <div id = "userFeedInnerCont">
                    <Suspense fallback={<Loader />}>
                        <div id = "userFeedInnerScroll">
                            {this.state.component}
                        </div>
                        <div id = "userFeedInputCont">
                            <textarea placeholder="Whats up? Type it here..."
                                      id = "userFeedTextArea"
                                      value = {this.state.textAreaValue}
                                      onChange = {this.onTextAreaValueChange}
                                      onKeyDown = {this.onTextAreaKeyDown}/>
                            <button id = "sendFeedBtn" onClick = {this.insertNewPost}>
                                Send
                            </button>
                        </div>
                    </Suspense>
                </div>
            </div>
        )
    }
}