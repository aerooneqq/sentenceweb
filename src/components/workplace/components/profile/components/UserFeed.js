import React, {Component, lazy, Suspense} from "react"
import "./styles/UserFeedStyles.css"

const UserAtomFeed = lazy(()=>import("./userFeedAtoms/UserAtomFeed"))
const UserFeedHeader = lazy(()=>import("./userFeedAtoms/UserFeedHeader"))
const Loader = lazy(() => import("../../../../loader/Loader"))

export default class UserFeed extends Component{ 
    constructor(props){ 
        super(props)
    }

    render(){ 
        return(
            <div id = "userFeedContainer"> 
                <UserFeedHeader />
                <div id = "userFeedInnerCont">
                    <Suspense fallback={<Loader />}>
                        <div id = "userFeedInnerScroll">
                            <UserAtomFeed />
                            <UserAtomFeed />
                            <UserAtomFeed />
                            <UserAtomFeed />
                        </div>
                        <div id = "userFeedInputCont">
                            <textarea placeholder="Whats up? Type it here..." id = "userFeedTextArea"/>
                            <button id = "sendFeedBtn">
                                Send
                            </button>
                        </div>
                    </Suspense>
                </div>
            </div>
        )
    }
}