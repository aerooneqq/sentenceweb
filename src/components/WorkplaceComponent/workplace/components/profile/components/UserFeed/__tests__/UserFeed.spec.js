import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import UserFeed from "../UserFeed";

describe("UserFeed", () => { 
    it("should render correctly", () => { 
        const userFeed = shallow(<UserFeed />);
        expect(shallowToJson(userFeed)).toMatchSnapshot();
    });

    it("should have proper initial state", () => { 
        const properState = { 
            isFirstLoad: true,
            textAreaValue: ""
        }
        const userFeed = shallow(<UserFeed />);

        for (let key in properState) { 
            expect(properState[key]).toBe(userFeed.state(key));
        }
    });
});