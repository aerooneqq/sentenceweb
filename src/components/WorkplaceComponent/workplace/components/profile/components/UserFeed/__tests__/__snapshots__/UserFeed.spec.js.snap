// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFeed should render correctly 1`] = `
<div
  id="userFeedContainer"
>
  <ProfileHeader
    header="Feed"
  />
  <div
    id="userFeedInnerCont"
  >
    <div>
      <div
        id="userFeedInnerScroll"
      >
        <UserFeedLoader />
      </div>
      <div
        id="userFeedInputCont"
      >
        <textarea
          id="userFeedTextArea"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Whats up? Type it here..."
          value=""
        />
        <button
          id="sendFeedBtn"
          onClick={[Function]}
        >
          Send
        </button>
      </div>
    </div>
  </div>
</div>
`;
